signature/policy_config.go

```diff
@@ -123,7 +123,7 @@ func (m *policyTransportsMap) UnmarshalJSON(data []byte) error {
 	// So, use a temporary map of pointers-to-slices and convert.
 	tmpMap := map[string]*PolicyTransportScopes{}
 	if err := paranoidUnmarshalJSONObject(data, func(key string) interface{} {
-		transport, ok := transports.KnownTransports[key]
+		transport, ok := registeredtransports.KnownTransports[key]
```

Comment:
This will need more work; otherwise an user who imports only a subset of transports would refuse to use a `policy.json` which defines policies for other transports.

I suppose unmarshaling and remarshaling a `policy.json` should always keep it unmodified, so we can’t just drop the sections for unknown transports; instead, `policyTransportScopesWithTransport` will need to be taught to deal with a `nil` transport, and silently accept any key. It seems that `requirementsForImageRef` does the right thing, or close enough to it, even in such a situation.

It bothers me a bit that we would accept any typo in transport names, though… perhaps we could add a `registeredtransports.AllExistingTransportsRegistered bool = false`, with the “all-transports” `transports.init()` seting it to `true`, and refusing unknown values if `AllExistingTransportsRegistered`? The downside is that that would make it impossible to add policies private transports not known to `containers/image/transports` for a single-application-use.